frontend/src/app/utils/dates/timeline.helper.ts

```diff
@@ -0,0 +1,155 @@
+import { addDays, addHours, addMonths, addWeeks, addYears, daysInWeek, differenceInHours, differenceInYears, endOfDay, endOfMonth, endOfWeek, endOfYear, startOfDay, startOfMonth, startOfWeek, startOfYear } from 'date-fns';
+
+import { DatePipe } from '@angular/common';
+import { TimelineScale } from '#types/tasks/timeline/enums/timeline-scale';
+import { TimelineSection } from '#types/tasks/timeline/timeline-section';
+
+export const getTimelineBoundaries = (currentDate: Date, scale: TimelineScale, pageNumber: number): TimelineSection => {
+    switch (scale) {
+    case TimelineScale.Day:
+        return {
+            start: startOfDay(addDays(currentDate, -1 * pageNumber / 3)),
+            end: endOfDay(addDays(currentDate, pageNumber / 3))
+        };
+    case TimelineScale.Week:
+        return {
+            start: startOfWeek(addWeeks(currentDate, -1 * pageNumber / 2)),
+            end: endOfWeek(addWeeks(currentDate, pageNumber / 2))
+        };
+    case TimelineScale.Month:
+        return {
+            start: startOfMonth(addMonths(currentDate, -1 * pageNumber / 2)),
+            end: endOfMonth(addMonths(currentDate, pageNumber / 2))
+        };
+    case TimelineScale.Year:
+        return {
+            start: startOfMonth(addYears(currentDate, -1 * pageNumber / 2)),
+            end: endOfMonth(addYears(currentDate, pageNumber / 2))
+        };
+    default:
+        throw new Error('Invalid scale');
+    }
+};
+
+export const getTimelineSections = (boundaries: TimelineSection, scale: TimelineScale) => {
+    if (scale !== TimelineScale.Year) {
+        return getTimelineSubsections(boundaries, scale + 1);
+    }
+    let sectionHours = 24 * 367;
```

Comment:
avoid magic numbers
add such rule to your linter

Reply:
Removed magic numbers. Eslint no-magic-numbers works pretty wierdly so I haven't added it, but I will keep in mind avoiding magic numbers